plaidml/bridge/openvino/src/plaidml_plugin/ops/avg_pool.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset1.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+
+namespace PlaidMLPlugin {
+
+static OpRegistration reg("avgpool", [](const Context& ctx) {
+  auto* layer = dynamic_cast<ngraph::opset1::AvgPool*>(ctx.layer);
+  IE_ASSERT(ctx.operands.size() == 1);
+  auto I = ctx.operands.at(0);
+  std::vector<int> strides;
+  for (auto stride : layer->get_strides()) {
+    strides.push_back(stride);
+  }
+  std::vector<int> kernel;
+  for (auto k : layer->get_kernel()) {
+    kernel.push_back(k);
+  }
+  auto pool_type = plaidml::op::PoolMode::AVG;
+  auto input_layout = plaidml::op::TensorLayout::NCX;
+  auto autopad_mode = to_plaidml(layer->get_auto_pad());
+  bool include_padding_in_avg = false;
+  bool use_ceil_for_output_shape = false;
+  std::vector<int> padding;
+  plaidml::edsl::Tensor result;
+  if (autopad_mode == plaidml::op::AutoPadMode::EXPLICIT) {
+    for (auto pad : layer->get_pads_begin()) {
+      padding.push_back(pad);
+    }
+    for (auto pad : layer->get_pads_end()) {
+      padding.push_back(pad);
+    }
+    result = op::pool(I, pool_type, kernel, strides, autopad_mode, padding, input_layout, include_padding_in_avg,
+                      use_ceil_for_output_shape);
+  } else {
+    result = op::pool(I, pool_type, kernel, strides, autopad_mode, padding, input_layout, include_padding_in_avg,
```

Comment:
Since this is identical to the code inside the true case of the if, I think we should put the common code just after the if.

So:

```
if (autopad_mode == plaidml::op::AutoPadMode::EXPLICIT) {
  for (auto pad : layer->get_pads_begin()) {
    padding.push_back(pad);
  }
  for (auto pad : layer->get_pads_end()) {
    padding.push_back(pad);
  }
}

auto result = op::pool(I, pool_type, kernel, strides, autopad_mode, padding, input_layout, include_padding_in_avg,
                      use_ceil_for_output_shape);
```

Reply:
Oh, yes. That looks cleaner. Thanks.